### PR TITLE
[SP-172] Caching package info

### PR DESCRIPTION
**Pull request by shawn-cx-li**
https://aussiecommerce.atlassian.net/browse/SP-172

### CODE REVIEW

File: src/scripts/indexOffers.ts [hunk 1]
```diff
@@ -141,6 +167,13 @@ export async function indexOffers(region: string): Promise<any> {
             offer.id_salesforce_external
           );
         }
+
+        for (const pkg of packageDetails) {
+          await redis.hmset(
```
Comment (sung-luxuryescapes) [inline src/scripts/indexOffers.ts:172]: nice. does `hmset` sets iterate through all fields and set them individually?

Comment (shawn-cx-li) [reply src/scripts/indexOffers.ts:172]: no, this is to iterate all packages under one offer, `hmset` should sets all field at once

**Comment by shawn-cx-li** [comment]
> looks good. one concern is that the [official Redis doc](https://redis.io/commands/keys) discourages the use of `keys`
> 
> > Warning: consider KEYS as a command that should only be used in production environments with extreme care. It may ruin performance when it is executed against large databases. This command is intended for debugging and special operations, such as changing your keyspace layout. Don't use KEYS in your regular application code. If you're looking for a way to find keys in a subset of your keyspace, consider using SCAN or sets.
> 
> Let's focus on building working solution first, but let's also think about how to design alternative solution without using `keys`.

Thanks for pointing out, agree to push `keys` command replacement for later

**Comment by sung-luxuryescapes** [comment]
actually rather than using `keys`, i think we can build our own hashmaps.

**Pull request merged**